### PR TITLE
ENT-3510: Fixed a couple of new compiler warnings

### PR DESCRIPTION
**Pull request by olehermanse**
Found using ubuntu 18 vagrant virtualbox VM (and gcc).

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins with exotics, please.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2465/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2465/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2465/

File: libpromises/crypto.c [hunk 1]
```diff
@@ -833,6 +841,8 @@ static void SetupOpenSSLThreadLocks(void)
 #ifndef __MINGW32__
     CRYPTO_set_id_callback((unsigned long (*)())ThreadId_callback);
 #endif
+    // This is a no-op macro for OpenSSL >= 1.1.0
+    // The callback function is not defined then
```
Comment (vpodzime) [inline libpromises/crypto.c]: *not **used** then*

Comment (olehermanse) [reply libpromises/crypto.c]: Both are correct, but I meant **defined**, as in, there is a guard around the definition as well.

Comment (olehermanse) [reply libpromises/crypto.c]: Clarified.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2470/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2470/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2470/

**Pull request merged**